File: Champions/Lucian/W.lua
Vector2 = require 'Vector2' -- include 2d vector lib 

function onStartCasting()
    local current = Vector2:new(owner.X, owner.Y)
    local to = (Vector2:new(spell.X, spell.Y) - current):normalize()
    local range = to * 900
    local trueCoords = current + range

    addProjectile("LucianWMissile", trueCoords.x, trueCoords.y)
end

function onFinishCasting()

end

function applyEffects()
    local damage = 20+spellLevel*40+owner:GetStats().AbilityPower.Total*0.9
    dealMagicalDamage(damage)
end

function onUpdate(diff)

end

File: Champions/Lucian/W.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Numerics;
using LeagueSandbox.GameServer.Logic.GameObjects;
using LeagueSandbox.GameServer.Logic.API;
using LeagueSandbox.GameServer.Logic.Scripting.CSharp;

namespace Lucian
{
    public class W : GameScript
    {
        public void OnActivate(Champion owner) { }
        public void OnDeactivate(Champion owner) { }
        public void OnStartCasting(Champion owner, Spell spell, Unit target){

        }
        public void OnFinishCasting(Champion owner, Spell spell, Unit target) {
            var current = new Vector2(owner.X, owner.Y);
            var to = Vector2.Normalize(new Vector2(spell.X, spell.Y) - current);
            var range = to * 900;
            var trueCoords = current + range;

            spell.AddProjectile("LucianWMissile", trueCoords.X, trueCoords.Y);
        }
        public void ApplyEffects(Champion owner, Unit target, Spell spell, Projectile projectile) {
            //float damage = 20 + (spellLevel * 40) + owner.GetStats().AbilityPower.Total * 0.9;
            //dealMagicalDamage(damage);
        }
        public void OnUpdate(double diff) {

        }
     }
}